internal/transactionsubmission/engine/signing/distribution_account_env_signature_client.go

```diff
@@ -0,0 +1,129 @@
+	}
+
+	return &DistributionAccountEnvSignatureClient{
+		networkPassphrase:   opts.NetworkPassphrase,
+		distributionAccount: distributionKP.Address(),
+		distributionKP:      distributionKP,
+	}, nil
+}
+
+var _ SignatureClient = (*DistributionAccountEnvSignatureClient)(nil)
+
+func (c *DistributionAccountEnvSignatureClient) SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error) {
+	if stellarTx == nil {
+		return nil, fmt.Errorf("stellarTx cannot be nil in %s", c.Type())
+	}
+
+	// Ensure that the distribution account is the only account signing the transaction
+	for _, stellarAccount := range stellarAccounts {
+		if stellarAccount != c.distributionAccount {
+			return nil, fmt.Errorf("stellar account %s is not allowed to sign in %s", stellarAccount, c.Type())
+		}
+	}
+
+	signedStellarTx, err = stellarTx.Sign(c.NetworkPassphrase(), c.distributionKP)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction in %s: %w", c.Type(), err)
+	}
+
+	return signedStellarTx, nil
+}
+
+func (c *DistributionAccountEnvSignatureClient) SignFeeBumpStellarTransaction(ctx context.Context, feeBumpStellarTx *txnbuild.FeeBumpTransaction, stellarAccounts ...string) (signedFeeBumpStellarTx *txnbuild.FeeBumpTransaction, err error) {
+	if feeBumpStellarTx == nil {
+		return nil, fmt.Errorf("stellarTx cannot be nil in %s", c.Type())
+	}
+
+	// Ensure that the distribution account is the only account signing the transaction
+	for _, stellarAccount := range stellarAccounts {
+		if stellarAccount != c.distributionAccount {
+			return nil, fmt.Errorf("stellar account %s is not allowed to sign in %s", stellarAccount, c.Type())
+		}
+	}
+
+	signedFeeBumpStellarTx, err = feeBumpStellarTx.Sign(c.NetworkPassphrase(), c.distributionKP)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction in %s: %w", c.Type(), err)
+	}
+
+	return signedFeeBumpStellarTx, nil
+}
+
+func (c *DistributionAccountEnvSignatureClient) BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error) {
+	return nil, fmt.Errorf("signature client of type %s does not support BatchInsert", c.Type())
+}
+
+func (c *DistributionAccountEnvSignatureClient) Delete(ctx context.Context, publicKey string) error {
+	return fmt.Errorf("signature client of type %s does not support Delete", c.Type())
+}
+
+func (c *DistributionAccountEnvSignatureClient) Type() string {
+	return string(SignatureClientTypeDistributionAccountEnv)
+}
+
+func (c *DistributionAccountEnvSignatureClient) NetworkPassphrase() string {
+	return c.networkPassphrase
+}
+
+var _ DistributionAccountResolver = (*DistributionAccountEnvSignatureClient)(nil)
+
+func (c *DistributionAccountEnvSignatureClient) DistributionAccount() string {
+	// var keyword string
+	// log.Warnf("Distribution accounts are not resolved by keyword in DistributionAccountEnvSignatureClient, keyword: '%s' is being ignored", keyword)
```

Comment:
Are these comments necessary?

Reply:
Removed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/174/commits/7a02ff18489ce2aa017b64f46bee7703f7f3e27d.